packages/sdk/contractkit/src/wrappers/BaseWrapper.ts

```diff
@@ -24,13 +28,40 @@ type EventsEnum<T extends Contract> = {
 
 /** Base ContractWrapper */
 export abstract class BaseWrapper<T extends Contract> {
+  protected _version?: T['methods'] extends ICeloVersionedContract['methods']
+    ? ContractVersion
+    : never
+
   constructor(protected readonly kit: ContractKit, protected readonly contract: T) {}
 
   /** Contract address */
   get address(): string {
     return this.contract.options.address
   }
 
+  async version() {
+    if (!this._version) {
```

Comment:
This fine now of course, but I think at some point we should probably add clearing caches to the wrappers